### PR TITLE
Reworks Tasers (doesn't give them back to Security though) Because Funny

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

https://github.com/user-attachments/assets/ef727913-6568-4705-91ca-330e7a1bb177

- Electrodes now have a physical representation in the form of a beam

- When you are hit with an electrode you will rapidly lose stamina while the beam is attached

- The electrode, being physical, can be interacted with
   - If you are stun immune, you can resist to remove them
   - If you get out of range of them, they'll remove themselves
   - You can help intent on someone to remove them
   - If someone runs into them, they will become tangled in them (being partially stunned themselves, but removing them) 

- Yes, this affects AI turrets. AI turret electrodes are stronger (they do double the amount of stamina damage) 
   - Turret tasers will avoid tasing the same target for infinite time, and will attempt to avoid tasing the same target

- Handheld tasers can be cancelled early by clicking on the target again

- Tasers will drain power as they continue to tase a target

## Why It's Good For The Game

Tasers are pretty one note so I thought it'd be cool to add some depth to them while making them function more like... well... a taser.

<Details>

<Summary> Original idea behind this </Summary>

I had an idea, in that we could give security more specialized gear setups (a la Goon) - putting stuff like tasers in a support weapon position while batons sit in a "front line" position, then allowing people to pick and choose at shift start

This way we could balance certain aspects of the security toolkit (*cough cough batons*) while still allowing newer players to be effective in a security position 

As right now, if you want to contribute as a security officer, you **have** to know your way around melee combat - lest you get disarmed and owned, which is why it's so strong. 

Buuuuut if newer players had the option to pick a primarily ranged option (with its own strengths and weaknesses), or an option that is most effective when paired with a friend, we could bring the baton strength down accordingly.

</Details>

## Changelog

:cl: Melbert
balance: Tasers are now more realistic
del: Electrodes are no longer in the hallucination projectile pool
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
does this change taser slugs and pulse rifle taser mode?

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 1]
```diff
@@ -1,29 +1,345 @@
...
+	owner.add_mood_event("tased", /datum/mood_event/tased)
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/being_tased)
+	if(!HAS_TRAIT(owner, TRAIT_ANALGESIA))
+		owner.emote("scream")
+	if(HAS_TRAIT(owner, TRAIT_HULK))
+		owner.say(pick(
+			";RAAAAAAAARGH!",
+			";HNNNNNNNNNGGGGGGH!",
+			";GWAAAAAAAARRRHHH!",
+			"NNNNNNNNGGGGGGGGHH!",
```
Comment (Sadboysuss) [inline code/modules/projectiles/projectile/energy/stun.dm:136]:
```suggestion
			";NNNNNNNNGGGGGGGGHH!",
```

**Comment by MrMelbert** [comment]
Those are in fact tasers

**Review by grungussuss — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/energy/stun.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/projectiles/projectile/energy/stun.dm:136]: That is untouched from existing tasers so I'm keeping it as-is

**Comment by necromanceranne** [comment]
YEAH BABY, YOU DID IT I'VE BEEN WAITING FOR THIS PR

**Comment by kuricityy** [comment]
give sec melberts tasers 2024..

**Comment by ilikesaying** [comment]
Turret-mounted taser harpoon that draws from placed cables when
(Me unleashing the full power of a Hotwired SM into the poor ruffians who broke into my department)

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 2]
```diff
@@ -2,28 +2,343 @@
...
+		return .
+	do_sparks(1, TRUE, src)
+	if(. == BULLET_ACT_BLOCK || !isliving(target) || blocked >= 100)
+		visible_message(span_warning("[src]\s fail to shock [target], falling to [get_turf(target)]."))
+		return .
+
+	var/mob/living/tased = target
+	if(tased.apply_status_effect(/datum/status_effect/tased, fired_from, firer, tase_stamina, null, "\the [src]\s"))
+		return .
+	visible_message(span_warning("[src]\s fail to shock [target], falling to [get_turf(target)]."))
```
Comment (necromanceranne) [inline code/modules/projectiles/projectile/energy/stun.dm:25]: I already noticed this in the video, but this probably parses very weirdly. Notably, it said '...falling to the reinforced wall.' when an electrode hit a wall.

Honestly, unless this has any mechanical relevancy, you may as well not have that last part.

**Comment by mc-oofert** [comment]
finally deathmatch taser nerf

File: code/modules/projectiles/projectile/energy/stun.dm (hunk 2 above)
Comment (SmArtKar) [reply code/modules/projectiles/projectile/energy/stun.dm:25]: There's ``last_tick_turf`` which could be used for this, but I agree with necro - i don't think that this is particularly necessary.

**Comment by deathrobotpunch** [comment]
give it a 1 precent chance to cause cardiac arrest that increases to 10 if you have a character at max age

**Comment by CCC23** [comment]
now we just give security a handgun with a 50 50 chance to shoot a taser or a 9mm bullet so you can get the true cop experience

**Comment by Sadboysuss** [comment]
> give it a 1 precent chance to cause cardiac arrest that increases to 10 if you have a character at max age

Aaaaand asimov AI can no longer be fully non-lethal

**Comment by ju45he** [comment]
Will this work with the chameleon gun or is it gonna just be awkward shooting some weird piece of a taser cord with it?

**Comment by MrMelbert** [comment]
> Will this work with the chameleon gun or is it gonna just be awkward shooting some weird piece of a taser cord with it?

I haven't tested it but given the behavior is tied to the type, I think it'll just be the electrode projectile like normal

> Edit: I think I remember seeing like this exact thing on monkestation before I got banned there.

Yeah they took it from me

**Comment by necromanceranne** [comment]
how does it interact with sleeping carp or other deflection effects?

**Comment by MrMelbert** [comment]
Good question

**Comment by MrMelbert** [comment]
Ok it actually uno reverses and makes the scarp user the taser-er so that's funny

**Pull request merged**